src/gordon_gcp/plugins/service/event_consumer.py

```diff
@@ -304,6 +312,21 @@ def _get_and_validate_pubsub_msg_schema(self, pubsub_msg_data):
             self._logger, {'msg_id': msg_id})
         msg_logger.info('Received new message.')
 
+        msg_datetime = pubsub_msg.publish_time
+        # need a non-TZ-aware object in UTC
```

Comment:
This comment isn't needed.  I'm all for using UTC everywhere we can, but you should be able to get a timedelta no matter what timezone it has.

Reply:
datetimes work in strange ways.  There are aware and non-aware ones.  The non-aware ones have to be created with a particular TZ, but then don't record the TZ.  They're giving us a non-aware UTC datetime, so that's what we need to compare with.

To clarify:
```
>>> import datetime
>>> datetime.datetime.now()
datetime.datetime(2018, 8, 7, 12, 31, 48, 589097)
>>> datetime.datetime.utcnow()
datetime.datetime(2018, 8, 7, 16, 31, 54, 83595)
>>> datetime.datetime.now(datetime.timezone.utc)
datetime.datetime(2018, 8, 7, 16, 32, 7, 868219, tzinfo=datetime.timezone.utc)
```

The first two are non-aware objects based on two different TZs.  The third is TZ-aware and therefore can be compared to TZ-aware objects in any TZ.

Reply:
(The docs call non-aware "naive" btw.)

Reply:
(And they won't let you operate on aware and naive objects in the same expression.)

Reply:
Thanks, I've never used datetime with timezones before, so I wasn't aware of this limitation. Using timezones should be the exception rather than without.